src/main/java/fi/aalto/cs/apluscourses/utils/DomUtil.java

```diff
@@ -118,9 +124,42 @@ private DomUtil() {
   public static Document parse(@NotNull InputStream stream) throws IOException, SAXException {
     try {
       return documentBuilderFactory.newDocumentBuilder().parse(stream);
+    } catch (ParserConfigurationException e) {
+      throw new IllegalStateException();
+    }
+  }
 
+  /**
+   * Parses the content of the given file to a DOM document.
+   * @param file A {@link File}.
+   * @return A DOM {@link Document}.
+   * @throws IOException  If the stream cannot be read.
+   * @throws SAXException If the content of the stream is not properly structured.
+   */
+  @NotNull
+  public static Document parse(@NotNull File file) throws IOException, SAXException {
+    try {
+      return documentBuilderFactory.newDocumentBuilder().parse(file);
     } catch (ParserConfigurationException e) {
       throw new IllegalStateException();
     }
   }
+
+  /**
+   * Writes the given document to the given file.
+   * @param document The document that is written to the file.
+   * @param out      The file to which the document is written.
+   * @throws IOException If an IO error occurs.
+   */
+  public static void writeDocumentToFile(@NotNull Document document, @NotNull File out)
+      throws IOException {
+    Source source = new DOMSource(document);
+    StreamResult result = new StreamResult(out);
+    try {
+      Transformer transformer = TransformerFactory.newInstance().newTransformer();
```

Comment:
Sonar points a vulnerability here (see SonarCloud report). Could we have a static field `transformerFactory` (cf. `documentBuilderFactory` in the same class) which is initialized with these two attributes
```java
transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");

transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, ""); 
```

Reply:
Done, good solution, thanks :+1: